src/3d/chunks/qgschunkedentity_p.cpp

```diff
@@ -171,12 +173,21 @@ void QgsChunkedEntity::update( const SceneState &state )
     }
   }
 
-  // unload those that are over the limit for replacement
-  // TODO: what to do when our cache is too small and nodes are being constantly evicted + loaded again
-  while ( mReplacementQueue->count() > mMaxLoadedChunks )
+  long long usedGpuMemory = 0;
+  for ( Qt3DRender::QBuffer *buffer : this->findChildren<Qt3DRender::QBuffer *>() )
+  {
+    usedGpuMemory += buffer->data().size();
+  }
```

Comment:
I don't really like this way of calculation. I would suggest to move that to a virtual function of QgsChunkedEntity, this code could be the default implementation, but other chunked entity classes should be able to do the memory size calculation differently - e.g. adding also size of textures...